vcd/resource_vcd_independent_disk.go

```diff
@@ -173,33 +208,212 @@ func resourceVcdIndependentDiskCreate(d *schema.ResourceData, meta interface{})
 
 	diskCreateParams.Disk.Description = d.Get("description").(string)
 
+	if value, ok := d.GetOk("sharing_type"); ok {
+		if vcdClient.Client.APIVCDMaxVersionIs("< 36.0") {
+			return diag.Errorf("`sharing_type` is supported from VCD 10.3+ version")
+		}
+		diskCreateParams.Disk.SharingType = value.(string)
+	}
+
 	task, err := vdc.CreateDisk(diskCreateParams)
 	if err != nil {
-		return fmt.Errorf("error creating independent disk: %s", err)
+		return diag.Errorf("error creating independent disk: %s", err)
 	}
 
 	err = task.WaitTaskCompletion()
 	if err != nil {
-		return fmt.Errorf("error waiting to finish creation of independent disk: %s", err)
+		return diag.Errorf("error waiting to finish creation of independent disk: %s", err)
 	}
 
 	diskHref := task.Task.Owner.HREF
 	disk, err := vdc.GetDiskByHref(diskHref)
 	if err != nil {
-		return fmt.Errorf("unable to find disk with href %s: %s", diskHref, err)
+		return diag.Errorf("unable to find disk with href %s: %s", diskHref, err)
 	}
 
 	d.SetId(disk.Disk.Id)
 
-	return resourceVcdIndependentDiskRead(d, meta)
+	return resourceVcdIndependentDiskRead(ctx, d, meta)
 }
 
-func resourceVcdIndependentDiskRead(d *schema.ResourceData, meta interface{}) error {
+func resourceVcdIndependentDiskUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	if d.HasChanges("size_in_mb", "storage_profile", "description") {
+		_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+		if err != nil {
+			return diag.Errorf(errorRetrievingOrgAndVdc, err)
+		}
+
+		storageProfileValue := d.Get("storage_profile").(string)
+		var storageProfileRef *types.Reference
+
+		if storageProfileValue != "" {
+			storageReference, err := vdc.FindStorageProfileReference(storageProfileValue)
+			if err != nil {
+				return diag.Errorf("error finding storage profile %s", storageProfileValue)
+			}
+			storageProfileRef = &types.Reference{HREF: storageReference.HREF}
+		}
+
+		disk, err := vdc.GetDiskById(d.Id(), true)
+		if err != nil {
+			return diag.Errorf("error fetching independent disk: %s", err)
+		}
+
+		diskAttachedVmsHrefs, err := disk.GetAttachedVmsHrefs()
+		if err != nil {
+			return diag.Errorf("error resourceVcdIndependentDiskUpdate faced issue fetching attached VMs")
+		}
+
+		// Lock on resource if independent disk is shared between more than one VM. This allows to avoid deadlock below
+		// when different independent shared disks try to acquire locks on same VMs
+		// Avoids situation like this:
+		// DiskA attached to VM2 and VM1 -> locked VM2, but can't lock VM1 because
+		// DiskB attached to VM1 and VM2 -> locked VM1, but can't lock VM2 (because it was locked by DiskA already)
+		if len(diskAttachedVmsHrefs) > 1 {
+			lockSharedDiskOpsGlobally()
+			defer unlockSharedDiskOpsGlobally()
```

Comment:
Oops! After we have moved this code around, lockSharedDiskOpsGlobally and unlockSharedDiskOpsGlobally are misnamed again. They don't know anything about "shared" disk anymore, so probably should be named:
lockIndependentDiskOpsGlobally and unlockIndependentDiskOpsGlobally
or just
lockDiskOpsGlobally and unlockDiskOpsGlobally